src/EventStore.Core/Services/PersistentSubscription/PersistentSubscriptionService.cs

```diff
@@ -718,24 +737,47 @@ string user
 				message.User?.Identity?.Name
 			);
 		}
-
-		private void RemoveSubscriptionConfig(string username, string eventSource, string groupName) {
+		
+		private void UpdateSubscriptionConfig(string username, string eventSource, string groupName, PersistentSubscriptionEntry replaceBy) {
 			_config.Updated = DateTime.Now;
 			_config.UpdatedBy = username;
 			var index = _config.Entries.FindLastIndex(x => x.Stream == eventSource && x.Group == groupName);
-			_config.Entries.RemoveAt(index);
+			
+			// Updating the persistent subscription if replaceBy is not null, otherwise delete the persistent subscription
+			if (replaceBy != null) {
+				_config.Entries[index] = replaceBy;
+			} else {
+				_config.Entries.RemoveAt(index);
+			}
```

Comment:
```suggestion
			// Update the persistent subscription if replaceBy is not null, otherwise delete the persistent subscription
			if (replaceBy != null)
				_config.Entries[index] = replaceBy;
			else
				_config.Entries.RemoveAt(index);
```